### ISSUE TITLE
🌳`E2E-E` support

### ISSUE THREAD

**Issue by Krutoy242** [roidrole/A-Dark-Mode-Addon#7]
Hello there, this is `E2E-E` dev!
I'm very excited about this dark mode and explicit `E2E-E` support. I want to make this texture pack official and shipped with modpack, or at least recommend it on modpack main page.

There is several incompats and I will post them in this thread when found.

Some for now:
- [ ] Scroll In creative menu  
  ![image](https://github.com/user-attachments/assets/e1489a96-710c-46f6-a3d3-cef55ac7232e)

- [ ] Just Enough Calculation
  ![image](https://github.com/user-attachments/assets/564585c9-0207-43f2-9fe7-7a22554202b3)

- [ ] Advanced Rocketry Machines (probably related to recent E2EE migrating to [fork](https://www.curseforge.com/minecraft/mc-mods/advanced-rocketry-2) mod?)
  ![image](https://github.com/user-attachments/assets/794b52d7-e14b-4b3d-ae0a-0c4c087732f1)

- [ ] Tinker's Complement Melter
  ![image](https://github.com/user-attachments/assets/fe99adc7-2883-41e3-9fc2-5dc2b40d8f82)

- [ ] Myrmex trades (probably [trading mod](https://www.curseforge.com/minecraft/mc-mods/gambling-style))
  ![image](https://github.com/user-attachments/assets/9ca5e698-0d33-4b8d-bb20-ecddcbaa289b)

- [ ] RFTools Modular Storage (And other RFTools machines)
  ![image](https://github.com/user-attachments/assets/9895ea02-6d7d-4993-b336-0e706d7e566a)

- [ ] Pam Harvest Craft Apiary
  ![image](https://github.com/user-attachments/assets/7eac4676-7895-42c7-9ee2-320bc26c4c7f)

- [ ] Forked Proxy Access
  ![image](https://github.com/user-attachments/assets/afa45878-e5b2-4355-9892-5fd2bded79e0)

- [ ] Colossal Chests
  ![image](https://github.com/user-attachments/assets/61785a9e-96e3-4198-afcf-73bc2d00fee6)

- [ ] Little Tiles
![image](https://github.com/user-attachments/assets/e6d0fb64-2a8d-4688-a96f-0cafce1875d2)

- [ ] Computer Craft Printer
![image](https://github.com/user-attachments/assets/98ae59ee-bcaf-4eca-9fbf-f10d505c10a9)

- [ ] IC2 Solar Arrays
![image](https://github.com/user-attachments/assets/c63164db-0701-4b39-9f56-7136af076e80)

**Comment by roidrole** [comment]
There are  few annoying white lines I haven't been able to fix with JECalc, but I will still publish once the other mods are supported with 1.8.0
RFTools doesn't seem to support resource packs for textures not already supported. I haven't seen any texture for them in RFTools' github. I might revisit later, but it seems outside of what a resource pack can do. Same goes for Little Tiles. 
I don't feel the need to remake the HarvestCraft's GUIs as they don't follow vanilla styles and are colored. That decision is up for debate

If these three first limitations are deal-breakers, I can remove the compatibility claim

**Comment by Krutoy242** [comment]
> If these limitations are deal-breakers, I can, at your request, remove the compatibility with E2EE claim

No, if you will maintain project and fix non-colored GUIs it will be awesome.
Also, technically, colores of programmically-drawn interfaces could be changed via CraftTweaker.

**Comment by roidrole** [comment]
> Also, technically, colores of programmically-drawn interfaces could be changed via CraftTweaker.

I'll probably investigate that possibility when I have more time, but it might take a while. Do you know where/if i can find docs for that?

**Comment by Krutoy242** [comment]
Docs for the Crafttweaker Mixins? Its very recent and complicated thing, i could only give an example of code E2EE using https://github.com/Krutoy242/Enigmatica2Expert-Extended/pull/407

**Comment by W4BEPMA** [comment]
![изображение](https://github.com/user-attachments/assets/7fb8000c-8cb5-4cf3-b915-1310be52f6db)
integrated terminals item storage tab

**Comment by W4BEPMA** [comment]
![изображение](https://github.com/user-attachments/assets/f0e532c3-c693-46a5-b1fb-f7a85046ee49)
crafting tab

**Comment by Krutoy242** [comment]
- [ ] `Extended Crafting` tables
  ![image](https://github.com/user-attachments/assets/d3854e8e-e4d4-45c7-9f28-e15ba0c8732e)

**Comment by Krutoy242** [comment]
- [ ] Schematica menu have white scroll
  ![image](https://github.com/user-attachments/assets/b526b902-f3bb-4722-af90-4d2aee47065a)

**Comment by Krutoy242** [comment]
- [ ] ME Interface
  ![image](https://github.com/user-attachments/assets/c4e89dfd-4b0c-4ce5-bcea-fb8eee841779)

**Comment by roidrole** [comment]
> Gendustry Item and Fluid gauges looks out of style. Same for slots of mod [Mechanics - Crafting Ways](https://www.curseforge.com/minecraft/mc-mods/mechanics-crafting-ways).
Have these been added by the mod itself or did you use a tweaker mod to create the JEI tabs?

**Comment by Krutoy242** [comment]
> Have these been added by the mods themselves

No, this JEI tabs linked on screenshot is Gendustry native ones. I didnt create them.

**Comment by roidrole** [comment]
All these, except for Schematica, should be fixed in 1.9.0 (pending CF approval)

**Comment by Krutoy242** [comment]
So, since @friendlyhj suggested to use ResourceModLoader, I can add it into E2EE if needed. Please instruct me if this necessary to load your dark mode scripts 😅

**Comment by roidrole** [comment]
I’m currently trying to make mixins work. For right now, I haven’t been able to, but I made **some** progress. For right now, there’s no benefit to using ResourceModLoader. I’ll update you if this changes

**Comment by Krutoy242** [comment]
For me, java launch argument `-Dmixin.debug.export=true` was a big helper.

**Comment by Krutoy242** [comment]
- [ ] Nutrition "N" hotkey menu
  ![image](https://github.com/user-attachments/assets/104c0a2b-7a24-43cf-8e23-d159a2193820)

**Comment by Krutoy242** [comment]
I have a question.
How could I make custom Replicator texture Dark Mode compatible? Right now it stores [there](https://github.com/Krutoy242/Enigmatica2Expert-Extended/blob/master/resources/enigmatica/textures/gui/replicator.png).

![image](https://github.com/user-attachments/assets/cb4a0198-19da-4e37-be27-b096bea788fd)

Could it be rewritten by resource pack?

**Comment by roidrole** [comment]
Not tested, but it seems it can in fact be rewritten by a resource pack. From Resource Loader’s CF page, 

![](https://i.imgur.com/tlO4kM7.png)

You (I’d) have to add it in the texture pack in assets/enigmatica/…

I’ll do more testing tonight

**Comment by roidrole** [comment]
After testing, resource packs can indeed overwrite resource loader's textures such as the replicator. I'll add this texture to 2.1.2. If you prefer making one yourself, I'll change it.
![replicator](https://github.com/user-attachments/assets/35aaa06a-be7f-4a48-94ce-c12d30a488d8)

**Comment by Krutoy242** [comment]
You can use PSD source file to more precisely change the color: https://github.com/Krutoy242/Enigmatica2Expert-Extended/blob/master/dev/resources/textures/replicator.psd

**Comment by BigBoss1941** [comment]
@roidrole The choice of text color has poor readability in my opinion, i highly suggest you keep some sort of contrast ...
![image](https://github.com/user-attachments/assets/db018a5f-0191-4d91-89b9-3a8d7a8841a7)

**Comment by Krutoy242** [comment]
In the next versions of E2EE i will add mod [AE2 Crafting Tree - Legacy](https://www.curseforge.com/minecraft/mc-mods/ae2-crafting-tree-legacy). Please add dark textures for it.
![image](https://github.com/user-attachments/assets/daf25ec9-1d2e-4aee-b129-6a449638542d)

**Comment by roidrole** [comment]
Crafting tree itself already has a dark mode - no need to add a separate compat for that. However, I did change the texture for the corner. Added to 2.2.0

**Comment by Krutoy242** [comment]
Quantum Compressor seems bit shifted 
![image](https://github.com/user-attachments/assets/dec9804f-66bd-43f7-b609-25067423c573)

**Comment by roidrole** [comment]
I somehow had already made that texture... regardless, fixed in 2.2.1!

**Comment by Krutoy242** [comment]
- [ ] Please add `nuclearcraft-1.12.2-2o.8.3` update support - new multiblocks added, E2EE is not updated yet, so you need to install it yourself.

![Image](https://github.com/user-attachments/assets/c37a0d34-a573-4144-80a9-9f83d97f92dc)

**Comment by Krutoy242** [comment]
- [ ] `Villager Market` from mod *Villager Market*

![Image](https://github.com/user-attachments/assets/44d64c9b-e443-4e36-9fe4-565a444ce8be)

**Comment by Krutoy242** [comment]
I just updated requious-frakto to the forked version: https://www.curseforge.com/minecraft/mc-mods/requious-frakto-unofficial
- [x] Please add support for this guis too

<img width="666" height="438" alt="Image" src="https://github.com/user-attachments/assets/7b5fa090-0c0d-4a03-89ee-d51f4f7f137a" />
